kloppy/domain/models/common.py

```diff
@@ -294,6 +302,79 @@ def duration(self):
         return self.end_timestamp - self.start_timestamp
 
 
+class Origin(Enum):
+    """
+    Attributes:
+        TOP_LEFT: Origin at the top left of the field
+        BOTTOM_RIGHT: Origin at the bottom left of the field
+        CENTER: Origin at the center of the field
+    """
+
+    TOP_LEFT = "top-left"
+    BOTTOM_RIGHT = "bottom-right"
+    CENTER = "center"
+
+    def __str__(self):
+        return self.value
+
+
+@dataclass
+class CoordinateSystem(ABC):
+    provider: Provider
+    origin: Origin
+    vertical_orientation: VerticalOrientation
+    normalized: bool
+    length: float = None
+    width: float = None
+
+    @property
+    @abstractmethod
+    def pitch_dimensions(self) -> PitchDimensions:
+        raise NotImplementedError
+
+
+@dataclass
+class KloppyCoordinateSystem(CoordinateSystem):
+    provider: Provider = None
+    origin: Origin = Origin.TOP_LEFT
+    vertical_orientation: VerticalOrientation = (
+        VerticalOrientation.TOP_TO_BOTTOM
+    )
+    normalized: bool = True
+    length: float = None
+    width: float = None
+
+    @property
+    def pitch_dimensions(self) -> PitchDimensions:
+        return PitchDimensions(
+            x_dim=Dimension(0, 1),
+            y_dim=Dimension(0, 1),
+            x_per_meter=1 / self.length,
+            y_per_meter=1 / self.width,
+        )
+
+
+@dataclass
+class TracabCoordinateSystem(CoordinateSystem):
+    provider: Provider = Provider.TRACAB
+    origin: Origin = Origin.CENTER
+    vertical_orientation: VerticalOrientation = (
+        VerticalOrientation.BOTTOM_TO_TOP
+    )
+    normalized: bool = False
+    length: float = None
+    width: float = None
+
+    @property
+    def pitch_dimensions(self) -> PitchDimensions:
+        return PitchDimensions(
+            x_dim=Dimension(-1 * self.length / 2, self.length / 2),
+            y_dim=Dimension(-1 * self.width / 2, self.width / 2),
+            x_per_meter=10000,
+            y_per_meter=10000,
+        )
```

Comment:
@koenvo Is there a better way to do this than by defining dataclasses? Ideally I'd like something like an Enum with properties, but not sure that is something possible.

Reply:
Well.. you could just pass a class around instead of an instance of it. That would work in some cases. But in this case you also have instance attributes (length, width) which are tied to a single instance of `TracabCoordinateSystem`.

This information is important as you can transform from TRACAB 105x68 to TRACAB 108x70 (not sure if dimensions make sense). If you use classes instead of instances it will act like a singleton and when you change attributes on one, the attributes will also be changed on the other (reference).

You could - at the bottom of the code for example - initialise some most-used versions.

```python
defaults = {
  Provider.TRACAB: TracabCoordinateSystem(normalized=False, length=105, ...)
}
```